enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,242 @@
+Customized annotations will contain a resource name requesting SR-IOV device.
+
+#### SR-IOV Operator
+
+- SR-IOV Operator will support new device type `switchdev` in Operator
+[SriovNetworkNodePolicy API](https://github.com/openshift/sriov-network-operator/blob/master/pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go#L33).
+
+- SR-IOV config daemon (a DaemonSet managed by SR-IOV Operator) will support
+changing e-switch mode from `legacy` to `switchdev` on the PF device and
+exposing VF representor information in `SriovNetworkNodeState` status.
+
+- SR-IOV network device plugin (a Device Plugin DaemonSet managed by SR-IOV
+Operator) supports advertising VF resource to kubelet.
+
+#### Machine Config Operator (MCO) [optional]
+
+- MCO will be used to create linux bond interface on two identical PFs that
+are capable of doing OVS hardware offload and create vlan interfaces on bond
+interface. The supported bond modes are `Active-Backup`, `Active-Active`
+and `LACP`. This is not required if OVS Hardware Offload is used on PF
+device directly.
+
+### User Stories
+
+#### Story 1
+
+Workflow of creating a SR-IOV pod using OVS hardware offload:
+- [optional] Deploy a baremetal cluster with API network on bonded PFs
+- Enable OVS hardware offload for OpenvSwitch instances via CNO
+- Install SR-IOV Operator from Operator Hub
+- Provision PFs to desired number of VFs via SR-IOV Operator
+- Create a pod specifying cluster network with customized net-attach-def
+
+### Implementation Details/Notes/Constraints
+
+- When enabling OVS Hardware Offload with SR-IOV VF LAG, PFs should first be
+provisioned with desired number of VFs, then configure PFs to `switchdev`
+mode, linux bond interface shall be created after above two steps. Since linux
+bond configuration can be applied via ignition file, it's important that SR-IOV
+Operator (installed in day 2) makes sure `switchdev` configurations are applied
+before linux bond configuration, this probably requires a node rebooting to
+guarantee the order.
+
+- When enabling OVS `hw-offload` option for OpenvSwitch daemon, it is required
+that ovsdb is created first.
+
+### Risks and Mitigations
+
+RHEL-8.3 is not used as base image for RHCOS when 4.6 gets released. RHEL-8.3
+contains kernel and driver changes to fully support OVS Hardware Offload with
+connection track feature.
+
+### Test Plan
+
+- The changes in each relevant component must pass full e2e suite.
+- OVS hardware offload functional tests must pass on supported NICs.
+
+### Graduation Criteria
+
+Tech Preview:
+- MCO supports configuring bond and vlan interfaces
+- Cluster Network Operator has API definition to enable OVS hardware offload
+- OVN-Kubernetes can configure VF and associated representor
+- Document how to use customized net-attach-def with Multus
+- SR-IOV Operator has API definition to configure `switchdev` device type
```

Comment:
This is very low-level, and it doesn't actually include the requirement "... and it actually works" :slightly_smiling_face:. I would just say that the criteria is that you can bring up a cluster using hardware offload, and you can do X, Y, and Z on it. (eg, you can bring up a mix of pods using offload and not using offload, on different nodes, and they can all communicate with each other, and you can apply NetworkPolicies to block some traffic, and that works too, ...)

Reply:
Updated